### ISSUE TITLE
Project refactor for improved development

### ISSUE THREAD

**Issue by AzorianMatt** [OthersideAI/self-operating-computer#88]
Hi there,

First and foremost, thank you for your effort on creating this! It has very good potential as shown by it's immediate popularity!

I have forked this project to start by cleaning up the code organization in efforts to make navigating the current solution much easier. Additionally, I am swapping the use of the `argparse` library in favor of `click` which has an overall more elegant approach and more feature-rich solution. I also intend to make some minor improvements such as automatic resolution detection among other things. Additionally, this comes with a very basic framework I have been using in projects for provide a standardized approach to app configuration through both file and environment based configurations very similar to what you already have done, but improved upon.

Would you be interested in merging this as a PR once I am finished? I can even look at merging some of the open PRs are that simple downstream so that you could by-pass that work as well. The progress may be viewed on my fork under the `feature-codebase-refactor` branch found [here](https://github.com/AzorianSolutions/self-operating-computer/tree/feature-codebase-refactor).

Thank you in advance for your consideration!

**Comment by michaelhhogue** [comment]
@AzorianMatt Thanks for this fork! Please keep in mind that we're trying to keep all of the code in main.py until it reaches 1,000+ lines, as mentioned in CONTRIBUTING.md.

However, this project structure could be very beneficial once that point is reached!

If you find click to be a better solution than argparse and find that it adds considerable value to the project without adding too many lines of code, definitely feel free to open a PR for it!

**Comment by AzorianMatt** [comment]
@michaelhhogue Thank you for the fast response!

Duly noted on the line total requirements, I didn't catch that when reading the contribution guidelines! My only thought on that and others may or may not share this sentiment, is that even at it's fairly short amount of lines currently, it still makes a bit difficult to navigate and learn. I'm one of those devs that read better with a structured system of files than a single flat source. Just something to consider perhaps in terms of reducing barrier to entry. Additionally, this approach is cleaning up a lot that lacks static references which we all know can ultimately lead to pesky bugs.

As far as click goes, it's another usability / barrier to entry thing where the automatic help prompts and overall feature set leads to a better outcome than argparse IMO. I started in Python using argparse, but quickly found it not as valuable once I discovered click.

I'll post updates here once I have my initial pass done, and leave it to you to evaluate if you find any value added.

Thanks again @michaelhhogue!
